### ISSUE TITLE
Error: [preflight] Drush::$container is not initialized yet. \Drush::setContainer() must be called with a real container.

### ISSUE THREAD

**Issue by tobiberlin** [drush-ops/drush#4659]
This is a follow up issue for https://github.com/drush-ops/drush/issues/4604 which was closed.

Here is the issue description:

> As I have this error only on one local Drupal / drush installation by now I am pretty sure it might be connected to this laptop - but I have no clue where to start / how to fix it. Maybe someone give me a hint.
> 
> Whenever I try to use drush I get the error message
> 
> [preflight] Drush::$container is not initialized yet. \Drush::setContainer() must be called with a real container.
> 
> I checked out our develop branch which is also checked out on our development server and tried the same on our development server. There drush runs as expected. I deleted the vendor folder where drush is installed and run composer install to get fresh files. No changes. I run composer update drush/drush but we have the latest version (10.3.6), Drupal is running on version 8.9.10, PHP 7.3.5 (laptop, local machine with WAMP) and 7.3.23 on development server. I have PHP 7.3.5 on another local machine where it still works.
> 
> On the same machine drush used to work - the problem just started today. Any ideas?

I now need to work on this laptop and tried to find the location where all fails. Before I started I completely deleted the vendor folder and re-installed everything with `composer install`.

It is `Drush\Config\Environment->getTmp()` where all fails. In line 139:

`$temporary_directory = Path::join(Drush::config()->cwd(), 'tmp');`

Here in `Drush::config()` the container is missing. There are 4 folders found and stored in the `$directories` variable within `getTmp()` but they are not writable so it comes to the line 139.

```
C:\Users\kraus\AppData\Local\Temp
C:\Users\kraus\AppData\Local\Temp
C:/WINDOWS/Temp
C:/Users/kraus/AppData/Local/Temp
```

As it is a Windows machine I am working on it may have to do with an update changing something on this folder (C:/Users/kraus/AppData/Local/Temp)?

**Comment by tobiberlin** [comment]
I start hating this laptop... I checked yesterday my other working Windows machine where the first of the above tmp folders was writable with the result that drush was working as it did not reach the line 139 (mentioned above). Now this morning I started my faulty laptop and did `drush cr` to check again - and now on this laptop the folder `C:/WINDOWS/Temp` is writable so that drush does not reach line 139 and everything is working. Apart from the magic happening on my laptop - may this code lline be a part of the code which is not reached usually because usually at least one of the system tmp folders is writable?! Shouldn't drush be able to handle such a situation as this code states?

**Comment by weitzman** [comment]
Can you make that folder writable and then run drush?

**Comment by tobiberlin** [comment]
I needed admin rights to change the write permission which I did not have. But changing the permission obviously solved the issue for me. Apart from this - I found at least one or two other issues about that where people described this error when the system tmp folder is not writable. Maybe it would be helpful to fire an exception instead of trying to call `Path::join(Drush::config()->cwd(), 'tmp');` whch does not work in this situation?!
